flink-runtime/src/test/java/org/apache/flink/runtime/execution/librarycache/BlobLibraryCacheManagerTest.java

```diff
@@ -585,6 +586,37 @@ public void releaseUserCodeClassLoader_willRegisterOnce()
         releaseHookLatch.await();
     }
 
+    @Test
+    public void testUseSystemClassLoader() throws IOException, InterruptedException {
+        final BlobLibraryCacheManager libraryCacheManager =
+                new TestingBlobLibraryCacheManagerBuilder().build();
+
+        final LibraryCacheManager.ClassLoaderLease classLoaderLease =
+                libraryCacheManager.registerClassLoaderLease(new JobID(), true);
+        UserCodeClassLoader userCodeClassLoader1 =
+                classLoaderLease.getOrResolveClassLoader(
+                        Collections.emptyList(), Collections.emptyList());
+
+        assertTrue(userCodeClassLoader1 instanceof SimpleUserCodeClassLoader);
```

Comment:
hmm....what kind of jobs actually don't have required jars? Does that also cover SQL jobs using a Flink connector provided in via lib?

If so, then this could break things if a (Flink) connector provided via lib uses a CL hook to be run some cleanup on job termination.

(I assume this change also covers cases where Flink connectors are being used)

Reply:
When users add flink connector jars to lib directory and setup a flink session cluster, jars and classpaths will always be empty if users submit SQL jobs without udf.

Thanks for pointing out the `CL hook` in the user class loader that I missed earlier, I will update the classloader in `JobManager` for `CL hook` when there are no jars and classpaths